using_images/db_images/mongodb.adoc

```diff
@@ -134,3 +134,91 @@ The MongoDB image can be run with mounted volumes to enable persistent storage f
 
 * *_/var/lib/mongodb_* - This is the database directory where
 MongoDB stores database files.
+
+=== Using the MongoDB replication
+
+IMPORTANT: The replication support provided by the MongoDB image is experimental and
+should not be used in production.
+
+This section describes how to start a cluster of MongoDB servers that implements
+master-slave replication and automated failover. This is the MongoDB’s
+recommended replication strategy.
+
+To start with MongoDB replication, you first have to define a DeploymentConfig
+that defines a replication controller which will manage the members of the
+MongoDB cluster. To tell MongoDB server that the member will be part of the
+cluster, you have to provide additional environment variables for the container
+defined in the replication controller pod template:
+
+[cols="3a,6a,1a",options="header"]
+|===
+
+|Variable name |Description |Default
+
+|`*MONGODB_REPLICA_NAME*`
+|Specify the name of the replication set.
+|rs0
+
+|`*MONGODB_KEYFILE_VALUE*`
+|See: http://docs.mongodb.org/manual/tutorial/generate-key-file[Generate a Key File]
+|generated
+|===
+
+When you create the DeploymentConfig and the pods with MongoDB cluster members
+will be started, they will not be initialized, but they will start as part of
+the `rs0` replication set.
+
+To initialize them, you will have to use a "service pod". This pod will start
+MongoDB with `initiate` argument, which will make the container entrypoint
+behave slighly different than having a regular standalone MongoDB database.
+
+The entrypoint will first discover all running member pods within the MongoDB
+cluster. For that, you have to use a 'headless' service named 'mongodb'. To define
+a headless service, you have to set the `portalIP` field value in the service
+definition to `None`. Then you can use a DNS query to get a list of the pod IP
+addresses that represents the current endpoints for this service.
+For example, you can use the `dig` command: `dig mongodb A +short +search`.
```

Comment:
I think a better way to approach documenting this all is to describe it like you did, but also say something like "the mongodb clustered service template sets this configuration here:" and then show the relevant json from the template.